### DISCUSSION TITLE
위키 추가 : local/Pull Request 템플릿

### DISCUSSION THREAD

**Discussion by greyHairChooseLife** [wanted-pre-onboarding-side-project/habit-tracker#33] [Suggestion]
## 근거
먼저 기존 레거시 이슈(아래에 링크)를 통해 일부 합의 된 부분이기도 하구요, 또 마침 어제 밤에 창환님이 말씀을 꺼내주셨듯이 서로의 코드를 설명하고 이해하기 위해 '코드' 외의 보조적인 수단이 필요하다는 데에 모두 공감하는 것 같습니다.

저 개인적으로 PR 마감 시간에 매번 쫓기듯이 개발하다가 PR 메시지를 급하게 후다닥 적게 된 경험이 많아요. 이렇게 급한 경우에도, 또 평소라고 해도 템플릿이 있어서 정해진 최소 작성 항목이 있다면 편하고 빠르게, 유익한 설명을 덧붙일 수 있을 것 같습니다.

## 제안

아래와 같은 형식의 템플릿 파일을 만들어서 PR 템플릿으로 쓰면 좋겠습니다.

그런데, 사실 많이 찾아본 것은 아니라 혹시 더하거나 수정 할 것이 있다면 적극적으로 말씀 해주시면 좋겠습니다.

```
1. 무슨 이유로 코드를 추가/변경 했는지
    :point_right: 관련 이슈를 링크 하되 간단히 요약해서 적어주기
    :point_right: close #issue

2. 파일 구조와 용도
    :point_right: `root/` 경로에서는 추가, 변경 된 파일들만 표시
    :point_right: `root/src/*` 이상에서는 모든 디렉토리와 파일들 표시
    :point_right: **추가** 된 파일은 아이콘, 이모지 등으로 추가 된 파일임을 표시
    :point_right: 용도/있어야 하는 이유 등을 간략히 적기

3. 어떤 부분에 리뷰어가 집중하면 좋을지

4. 관련 스크린샷 or GIF
```

## 참고
1. `local` 구분을 둔 것은 `local/` :point_right: `upstream/`의 경우로 한정하기 위해서 입니다.
2. 관련 [레거시 이슈](https://github.com/wanted-pre-onboarding-side-project/habit-tracker/issues/8)
3. [참고 블로그](https://2jinishappy.tistory.com/337) (내용이 짧고 좋은 것 같아요. 읽어 보시기를 추천 드립니다. 1분 미만)

**Comment by jiyeon2** [comment]
제안해 주신 내용 좋습니다~~

다만 2번 항목 '파일 구조와 용도'는 템플릿 항목으로 존재하더라도 
쓰고 싶으면 쓰고, 아니면 지워도 되는 옵셔널한 항목이면 좋겠습니다

전체적 파일 구조 리팩토링 작업의 경우 
추가, 변경된 파일을 일일이 구분하여 기재하는 과정이 번거로울 것으로 생각되며, 
추가되거나 변경된 파일과 구조는 리뷰화면에서 확인이 가능하기 때문입니다
![image](https://user-images.githubusercontent.com/18395475/231238891-5a23c322-264a-4a3d-b08c-23df471af294.png)

제가 #14 pr에서 파일디렉토리를 포함하여 작성하였는데
제 코드를 처음 접하게 되는 상황이었으므로
'저는 처음에 디렉터리를 이런식으로 구분하여 시작했습니다'를 보여드릴 의도였습니다
차후 pr에서는 디렉토리에 큰 변동이 없는 이상(디렉토리 변경이 주안점이 아닌 이상) 계속 작성할 생각은 없었습니다

**Reply by greyHairChooseLife** [reply]
파일 구조 관련해서,
구조 자체보다도 각각의 디렉토리와 파일의 쓰임새(?) 용도(?)를 간단하게 써 주자는 말이었습니다. 

말씀 듣고 보니 이게 자주 바뀔 수 있는데 참 번거로울 수 있을 것 같네요.

일단 저는 최하단에 항목을 두고 개인적으로 노력해서 써 보던가 할게요. 첫 PR과 리뷰 이후에 제 PR메시지를 읽어 보신 여러분의 생각을 다시 한번 여쭤보겠습니다. 도움이 됐는지를요.

@chadseok 
창환님 관련 의견 남겨 주시면 내용을 위키에 남기고 템플릿 설정을 해 두겠습니다.

**Reply by jiyeon2** [reply]
아 "`root/src/*` 이상에서는 모든 디렉토리와 파일들 표시" 라는 문구를 보고
제가 전체 디렉토리 구조를 기재해야 한다고 이해해서
상연님이 의도하신 것과 다르게 받아들였네요~

지금 설정하는 템플릿을 큰 틀로 생각하고
각자 작성한 첫 pr 메시지를 보며 더 자세히 이야기를 나눠봐도 좋겠습니다!

**Reply by chadseok** [reply]
네 저도 @greyHairChooseLife 님이 작성하신 PR에 들어갈 내용들 모두 필요하다고 생각이 듭니다!!
네. 템플릿으로 만드셔도 괜찮을 것 같아요. 다만 위에서 @jiyeon2 님이 말씀하셨다 싶이 폴더 구조 관련해서는 저도 직접 적용해보면서 피드백 드릴께요!!

**Reply by greyHairChooseLife** [reply]
좋습니다. 그러면 이 디스커션은 일단 열어둔 채로 두고 첫 PR을 마친 뒤 마무리 해요.
